solar_tracker.py

```diff
@@ -0,0 +1,209 @@
+            return "right"
+        return "stop"
+
+    def night_time_mode(self, ldr_values):
+        """
+        Mode which will make the software pause when there isn't enought light
+
+        ...
+
+        Compares both LDR values to 100, if both are lower than the value, stops
+        the linear actuators and enters into the night mode, which will make new
+        requests every 5 minutes in order to verify if there is enough light to
+        come back to work.
+
+        Parameters
+        ----------
+        ldr_values: list
+            Contains the LDR values sent by the Arduino
+
+        Returns
+        -------
+            None
+
+        -------
+
+        """
+
+        if ldr_values[0] < 100 and ldr_values[1] < 100:
+            self.controller.move("stop", self.output_pins)
+            if self.night_mode is False:
+                logger.info("Entering night mode...")
+                self.night_mode = True
+            sleep(300)
+            return None
+        elif self.night_mode:
+            logger.info("Exiting night mode!")
+            self.night_mode = False
+            return None
+
+    def run(self, strategy="empty"):
+        """
+        Executes the solar tracking, activating the linear actuators to move the
+        solar panel array
+
+        ...
+
+        Every 0.5 seconds, requests a new list of data from the Arduino,
+        process and validates it. If the LDR values are valid, verifies to
+        which direction the solar panel array should move and, if necessary,
+        moves it to that direction.
+        If the LDR values are not valid, stops any movement in order to keep
+        the precision of the software.
+
+        Parameters
+        ----------
+        strategy: string
+            Which strategy will be used to process the LDR data.
+            Default=empty
+
+        Exceptions
+        ----------
+        unvalid strategy:
+            Strategy not available in the software
+        IOError, OSError:
+            Errors hardware related, can be ignored and the software will
+        run normally
+
+        """
+        if strategy not in self.strategies.keys():
+            # Verifies if the strategy is valid
+            raise Exception("Invalid strategy!")
+        while True:
+            try:
+                sleep(0.5)
+                ldr_values = self.i2c.get_ldr_values()
+          
+            except (IOError, OSError) as e:
+                logger.warning(str(e))
+                continue
+        
+            except (OddLDRListException, EmptyLDRListException, UnvalidLDRListValuesException) as e:
+                self.controller.move("stop", self.output_pins)
+                logger.warning(str(e))
+                continue
+            
+            # Gets the selected strategy from the strategies dict
+            # and pass the ldr_values as an argument
+            if strategy == "empty":
```

Comment:
A função da estratégia empty tem que ter a mesma aridade das outras. Ta estranho isso. A estratégia deve receber o argumento e não fazer nada com ele se for o caso.